configs/components/ruby-3.2.5.rb

```diff
@@ -94,7 +94,7 @@
   special_flags = " --prefix=#{ruby_dir} --with-opt-dir=#{settings[:prefix]} "
 
   if settings[:supports_pie]
-    special_flags += " CFLAGS='#{settings[:cflags]}' LDFLAGS='#{settings[:ldflags]}' CPPFLAGS='#{settings[:cppflags]}' "
+    special_flags += " CFLAGS='-fPIC -fstack-protector-strong -fno-plt -O2' LDFLAGS='#{settings[:ldflags]}' CPPFLAGS='-fPIE #{settings[:cppflags]}' "
```

Comment:
Can you describe why you're removing :cflags from here? What problem is being solved aside from "pie doesn't work"? Does this mean none of the `cflags` defined in the project are being passed to the `configure` command?